importer/helpers/dagbuilder.go

```diff
@@ -12,27 +13,33 @@ import (
 	pb "github.com/TRON-US/go-unixfs/pb"
 
 	chunker "github.com/TRON-US/go-btfs-chunker"
-	cid "github.com/ipfs/go-cid"
 	files "github.com/TRON-US/go-btfs-files"
+	cid "github.com/ipfs/go-cid"
 	pi "github.com/ipfs/go-ipfs-posinfo"
 	ipld "github.com/ipfs/go-ipld-format"
 )
 
-var ErrMissingFsRef = errors.New("missing file path or URL, can't create filestore reference")
+var (
+	ErrMissingFsRef           = errors.New("missing file path or URL, can't create filestore reference")
+	ErrUnknownNodeType        = errors.New("unknown node type")
+	ErrUnexpectedNodeType     = errors.New("unexpected node type")
+	ErrUnexpectedProgramState = errors.New("unexpected program state occurred")
+)
 
 // dagBuilderHelper contains the shared fields among various different helpers
 // under the same DAG.
 type dagBuilderHelper struct {
-	dserv              ipld.DAGService
-	spl                chunker.Splitter
-	recvdErr           error
-	rawLeaves          bool
-	nextData           []byte // the next item to return.
-	maxlinks           int
-	cidBuilder         cid.Builder
-	tokenMetadata      []byte
-	tokenMetaToProcess bool   // Need to add the tokenMetadata to the current UnixFs node?
-	metaOffset         uint64 // starting offset of token metadata within the data block
+	Dserv      ipld.DAGService
+	spl        chunker.Splitter
+	recvdErr   error
+	rawLeaves  bool
+	nextData   []byte // the next item to return.
+	maxlinks   int
+	cidBuilder cid.Builder
+
+	metaDb       *MetaDagBuilderHelper
+	metaDagBuilt bool
+	reedSolomon  bool
```

Comment:
See comment above, this flag is not necessary, use `IsMultiDagBuilder()` to check for reed solomon case.